core/application/impl/kagome_application_impl.hpp

```diff
@@ -54,14 +55,15 @@ namespace kagome::application {
      * @param kagome_config kagome configuration parameters
      * @param keys_config keys parameters
      */
-    KagomeApplicationImpl(const std::string &config_path,
-                          const std::string &keystore_path,
-                          const std::string &leveldb_path,
-                          uint16_t p2p_port,
-                          uint16_t rpc_http_port,
-                          uint16_t rpc_ws_port,
-                          bool is_genesis_epoch,
-                          uint8_t verbosity);
+    KagomeApplicationImpl(
+        const std::string &config_path,
+        const std::string &keystore_path,
+        const std::string &leveldb_path,
+        uint16_t p2p_port,
+        const boost::asio::ip::tcp::endpoint &rpc_http_endpoint,
+        const boost::asio::ip::tcp::endpoint &rpc_ws_endpoint,
+        bool is_genesis_epoch,
+        uint8_t verbosity);
```

Comment:
Maybe would be better to extract to a struct ApplicationInjectorConfig?

Reply:
I'm not sure. We will to have just one application. Another ones exist only for test/demo, etc.

Reply:
I mean, just to avoid 8 arguments. It would allow, for example, named initialization, as structures support them, while function arguments not.

Reply:
Ok, I got it right. But i think better to make it in special PR. I'll suggest it for team.

Reply:
Makes sense. There might be structs like `ValidatingNodeConfig` or `FullSyncingNodeConfig`